### ISSUE TITLE
Loading json into a pg column using pgloader

### ISSUE THREAD

**Issue by vijmen** [dimitri/pgloader#250]
Hi Dimitri
I have a table with some columns of jsonb data type. I am trying to insert data (which I tested using direct SQL) using the working CSV template in the docs page. I get an error which I can't figure out.
Can you please help?
Below is the data, code & error details.
DATA:
```
1234|loc1|locl|addr1||market|stat1|country|pc|str1|ph1|em1|{"a":"7:00 AM - 12:00 AM"}|[]|121.432591|-23.18544375
5678|loc2|locl|addr1||market|stat1|country|pc2|str2|ph2|em3|{"a":"7:00 AM - 10:00 PM"}|["typ1"]|121.123458|-22.123456
```

SOURCE:
```
LOAD CSV  
        FROM '/home/ubuntu/loaders/ttest.csv' 
             (  
              c1                        ,
              c2                 ,
              c3                    ,
              c4            ,
              c5            ,
              c6                  ,
              c7                     ,
              c8                   ,
              c9                  ,
              c10                   ,
              c11                     ,
              c12                     ,
              c13             ,
              c14                ,
              c15                 ,
              c16                  
             )    
        INTO postgresql://db?table  
             (  
              c1             ,
              c2      ,
              c3         ,
              c4 ,
              c5 ,
              c6       ,
              c7          ,
              c8        ,
              c9       ,
              c10        ,
              c11          ,
              c12          ,
              c13  ,
              c14     ,
              location point using (format nil "(~a,~a)" c15 c16)
             )  
       WITH TRUNCATE,  
           fields optionally enclosed by '"',  
           fields escaped by double-quote,  
           fields terminated by '|'  
 
       SET    work_mem to '12MB',  
           standard_conforming_strings to 'on'  

;
```

ERROR LOG:

```
2015-06-25T07:09:16.047000+03:00 LOG Main logs in '/tmp/pgloader/pgloader.log'
2015-06-25T07:09:16.051000+03:00 LOG Data errors in '/tmp/pgloader/'
2015-06-25T07:09:16.051000+03:00 LOG Parsing commands from file #P"/home/ubuntu/loaders/xyz.load"
Heap exhausted during allocation: 1294598144 bytes available, 2818572304 requested.
 Gen StaPg UbSta LaSta LUbSt Boxed Unboxed LB   LUB  !move  Alloc  Waste   Trig    WP  GCs Mem-age
   0:     0     0     0     0     2     0     0     0     0        0 65536 42949672    0   0  0.0000
   1:     0     0     0     0   185    16     7 80644 80678 2648986352 371984  2000000  158   0  0.2020
   2:     0     0     0     0     0     0     0     0     0        0     0  2000000    0   0  0.0000
   3:     0     0     0     0     0     0     0     0     0        0     0  2000000    0   0  0.0000
   4:     0     0     0     0     0     0     0     0     0        0     0  2000000    0   0  0.0000
   5:     0     0     0     0     0     0     0     0     0        0     0  2000000    0   0  0.0000
   6:     0     0     0     0  3441  1088     0     0     0 148406272     0  2000000 3341   0  0.0000
   Total bytes allocated    = 2797392624
   Dynamic-space-size bytes = 4294967296
GC control variables:
   *GC-INHIBIT* = false
   *GC-PENDING* = true
   *STOP-FOR-GC-PENDING* = false
                    table name       read   imported     errors            time
------------------------------  ---------  ---------  ---------  --------------
                         fetch          0          0          0          0.008s
------------------------------  ---------  ---------  ---------  --------------
        db.x_test          0          0          1         22.661s
------------------------------  ---------  ---------  ---------  --------------
             Total import time          0          0          1         22.669s
2015-06-25T07:09:38.908000+03:00 ERROR A SB-KERNEL::HEAP-EXHAUSTED-ERROR condition without bindings for heap statistics.  (If
you did not expect to see this message, please report it.
```

Thanks
v

**Comment by dimitri** [comment]
You can't use double quote character for both quoting the outer CSV data and the inner JSON data, so I got some luck by using those options:

```
           fields optionally enclosed by '0x39',
           fields escaped by backslash-quote,  
           fields terminated by '|'  
```

Also it appears that some CSV parser errors where muffled by pgloader so I am adding some fix here and get:

```
2015-06-25T10:24:39.203000+02:00 ERROR we are reading non quoted csv data and found a quote at 97
1234|loc1|locl|addr1||market|stat1|country|pc|str1|ph1|em1|{"a":"7:00 AM - 12:00 AM"}|[]|121.432591|-23.18544375
2015-06-25T10:24:39.204000+02:00 DATA < #("5678" "loc2" "locl" "addr1" NIL "market" "stat1" "country" "pc2" "str2"
    "ph2" "em3" "{\"a\":\"7:00 AM - 10:00 PM\"}" "[\"typ1\"]"
    "(121.123458,-22.123456)")
```

I couldn't reproduce the heap exhaustion here, it might be because I am using a more recent cl-csv librairy that you are, so it might be worth building from sources (and if using the debian package, first `apt-get remove cl-*`).

**Comment by vijmen** [comment]
Thanks for that. I had version 2.3.3 running; so, I uninstalled and did a build from source.
However, I still get similar errors. To test whether this was a JSON issue, I removed the json elements and made them [].
It seems to be complaining about the numbers.
If you look at the error message you posted above, 
4:39.203000+02:00 ERROR we are reading non quoted csv data and found a quote at 97
1234|loc1|locl|addr1||market|stat1|country|pc|str1|ph1|em1|{"a":"7:00 AM - 12:00 AM"}|[]|121.432591|-23.18544375

character at 97 is a number.
So, am not sure if this is related to the JSON data.
